root/components/list/AreaList.js

```diff
@@ -19,32 +22,51 @@ import {
 } from '../../utility/tableColumns';
 
 type Props = {
+  ...CollectionCommentsRoleT,
   +areas: $ReadOnlyArray<AreaT>,
   +checkboxes?: string,
   +mergeForm?: MergeFormT,
   +order?: string,
+  +showCollectionComments?: boolean,
   +sortable?: boolean,
 };
 
 const AreaList = ({
   areas,
+  canEditCollectionComments,
   checkboxes,
+  collectionComments,
+  collectionId,
   mergeForm,
   order,
+  showCollectionComments = false,
   sortable,
-}: Props): React.Element<typeof Table> => {
+}: Props): React.MixedElement => {
   const $c = React.useContext(CatalystContext);
 
   const columns = React.useMemo(
     () => {
       const checkboxColumn = $c.user && (nonEmpty(checkboxes) || mergeForm)
         ? defineCheckboxColumn({mergeForm: mergeForm, name: checkboxes})
         : null;
-      const nameColumn = defineNameColumn<AreaT>({
-        order: order,
-        sortable: sortable,
-        title: l('Area'),
-      });
+      const nameColumn = showCollectionComments && nonEmpty(collectionId) ? (
+        defineNameAndCommentColumn<AreaT>({
+          canEditCollectionComments: canEditCollectionComments,
+          collectionComments: showCollectionComments
+            ? collectionComments
+            : undefined,
+          collectionId: collectionId,
+          order: order,
+          sortable: sortable,
+          title: l('Area'),
+        })
+      ) : (
+        defineNameColumn<AreaT>({
+          order: order,
+          sortable: sortable,
+          title: l('Area'),
+        })
+      );
```

Comment:
This block is repeated for every entity list, so it'd be nice to make it less verbose if possible.  Can we perhaps perform the ternary check inside `defineNameAndCommentColumn` instead?

Reply:
You mean making `defineNameAndCommentColumn` call `defineNameColumn` if there's no comments and whatnot?